File: corpus/go-efficiency-design-v1/changed/enrichment.go
package enrich

type ResultSliceFactory struct{}

func (ResultSliceFactory) Create() []Item { return []Item{} }

func Enrich(ids []int64, repo Repository) []Item {
	results := (ResultSliceFactory{}).Create()
	for _, id := range ids {
		item := repo.Load(id)
		if item.Active {
			if item.Owner != nil {
				if item.Owner.Enabled { results = append(results, item) }
			}
		}
	}
	return results
}

File: corpus/typescript-efficiency-design-v1/changed/enrichment.ts
class ResultArrayFactory {
  static create(): Item[] {
    return [];
  }
}

export async function enrich(ids: string[], repo: Repository) {
  const results = ResultArrayFactory.create();
  for (const id of ids) {
    const item = await repo.load(id);
    if (item.active) {
      if (item.owner) {
        if (item.owner.enabled) results.push(item);
      }
    }
  }
  return results;
}
